internal/controller/networkvalidator_controller.go

```diff
@@ -95,7 +96,7 @@ func (r *NetworkValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		ValidationRuleErrors:  make([]error, 0, vr.Spec.ExpectedResults),
 	}
 
-	networkService := validators.NewNetworkService(r.Log)
+	networkService := validators.NewNetworkService(&http.Client{}, r.Log)
```

Comment:
Using `http.Client{}` is not ideal. Typically, `http.DefaultClient` is preferable. However, in this case, we need more than just the default client. Please instantiate an HTTP client that loads the local system root CAs and includes a configurable option to disable TLS verification.

Reply:
See https://github.com/spectrocloud/ambit/blob/master/services/hubble/hubble_certs.go#L10

Reply:
And https://github.com/spectrocloud/ambit/blob/master/services/hubble/hubble_certs.go#L44